state-surgery/state/state.go

```diff
@@ -0,0 +1,196 @@
+package state
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"reflect"
+	"strings"
+
+	"github.com/ethereum-optimism/optimism/l2geth/common/hexutil"
+	"github.com/ethereum-optimism/optimism/state-surgery/solc"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// StorageValues represents the values to be set in storage.
+// The key is the name of the storage variable and the value
+// is the value to set in storage.
+type StorageValues map[string]any
+
+// EncodedStorage represents the storage key and value serialized
+// to be placed in Ethereum state.
+type EncodedStorage struct {
+	Key   common.Hash
+	Value common.Hash
+}
+
+// EncodedStorage will encode a storage layout
```

Comment:
Going to follow up with a PR that adds handling of offsets + mappings, those are the only required things to be able to fully run on our L2 contracts based on the output of this script:

```bash
#!/bin/bash

for file in contracts/L2/*; do
    fqn="$file:$(basename $file | cut -d '.' -f1)"
    forge inspect $fqn storageLayout
done
```

We need to set a mapping in the `L2CrossDomainMessenger` and one of the contracts has a variable with an offset